sys/shell/shell.c

```diff
@@ -454,11 +481,31 @@ static int readline(char *buf, size_t size)
 void shell_run_once(const shell_command_t *shell_commands,
                     char *line_buf, int len)
 {
+    if (IS_USED(MODULE_SHELL_LOCK)) {
+        puts("shell: shell_lock is an experimental feature and only shows as "
+             "a PoC how the shell could be protected with a password. Don't "
+             "expect relevant security from it for production, as "
+             "Man-in-the-Middle attacks are possible depending on the used "
+             "connection method!");
```

Comment:
let's move that to the test application

Reply:
Are you sure about that? People who try this module directly and not by running the test themself will never get the warning then.